query/src/storages/fuse/operations/commit.rs

```diff
@@ -33,18 +36,65 @@ use crate::storages::fuse::operations::TableOperationLog;
 use crate::storages::fuse::statistics;
 use crate::storages::fuse::FuseTable;
 use crate::storages::fuse::TBL_OPT_KEY_SNAPSHOT_LOC;
+use crate::storages::Table;
 
 impl FuseTable {
-    #[inline]
     pub async fn do_commit(
         &self,
         ctx: Arc<QueryContext>,
         operation_log: TableOperationLog,
         overwrite: bool,
     ) -> Result<()> {
-        // TODO OCC retry & resolves conflicts if applicable
+        // TODO : configuration of backoff strategy
+        let mut backoff = ExponentialBackoff::default();
+        let tid = self.table_info.ident.table_id;
+        let mut tbl = self;
+        let mut latest: Arc<dyn Table>;
+        loop {
+            match tbl
+                .try_commit(ctx.as_ref(), &operation_log, overwrite)
+                .await
+            {
+                Ok(_) => break Ok(()),
+                Err(e) if e.code() == ErrorCode::table_version_mismatched_code() => {
```

Comment:
It would be better to add a `tracing::error!` to log the re-try error here

Reply:
ok